File: pac4j-async-core/src/test/java/org/pac4j/async/core/TestCredentials.java
package org.pac4j.async.core;

import org.pac4j.core.credentials.Credentials;

/**
 * Quick demo credentials. Note that these are not
 */
public class TestCredentials extends Credentials {

    private final String name;
    private final String password;

    public TestCredentials(String name, String password) {
        this.name = name;
        this.password = password;
    }

    public String getName() {
        return name;
    }

    public String getPassword() {
        return password;
    }

    @Override
    public boolean equals(Object o) {

        if (! (o instanceof TestCredentials)) {
            return false;
        }

        final TestCredentials other = (TestCredentials) o;

        return this.name == other.name && this.password == other.password;
    }

    @Override
    public int hashCode() {
        int result = name.hashCode();
        result = 31 * result + password.hashCode();
        return result;
    }
}

File: vertx-pac4j-async-demo/src/test/kotlin/org/pac4j/async/vertx/TestCredentials.kt
package org.pac4j.async.vertx

import org.pac4j.core.credentials.Credentials

/**
 *
 */
class TestCredentials(val name: String, val password: String): Credentials() {

    override fun equals(o: Any?): Boolean {

        if (o !is TestCredentials) {
            return false
        }

        val other = o as TestCredentials?

        return this.name === other!!.name && this.password === other!!.password
    }

    override fun hashCode(): Int {
        var result = name.hashCode()
        result = 31 * result + password.hashCode()
        return result
    }


}